searchlib/src/vespa/searchlib/queryeval/nearest_neighbor_blueprint.cpp

```diff
@@ -73,20 +74,31 @@ NearestNeighborBlueprint::NearestNeighborBlueprint(const queryeval::FieldSpec& f
         _dist_fun = nns_index->distance_function();
     }
     uint32_t est_hits = _attr_tensor.getNumDocs();
-    if (_approximate && nns_index) {
-        est_hits = std::min(target_num_hits, est_hits);
-    }
     setEstimate(HitEstimate(est_hits, false));
     set_want_global_filter(true);
 }
 
 NearestNeighborBlueprint::~NearestNeighborBlueprint() = default;
 
 void
-NearestNeighborBlueprint::set_global_filter(std::shared_ptr<BitVector> global_filter)
+NearestNeighborBlueprint::set_global_filter(const GlobalFilter &global_filter)
```

Comment:
We should create a unit test for what happens when the global filter is considered. Should be sufficient to test the estimate result on the blueprint, and whether we are still approximate or not.